test/integration/apiserver/admissionwebhook/mutating_webhook_gvk_conversion_test.go

```diff
@@ -0,0 +1,398 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/integration/etcd"
+
+	clientset "k8s.io/client-go/kubernetes"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+var (
+	runtimeSchemeGVKTest = runtime.NewScheme()
+	codecFactoryGVKTest  = serializer.NewCodecFactory(runtimeSchemeGVKTest)
+	deserializerGVKTest  = codecFactoryGVKTest.UniversalDeserializer()
+)
+
+type admissionTypeChecker struct {
+	mu       sync.Mutex
+	upCh     chan struct{}
+	upOnce   sync.Once
+	requests []*admissionv1.AdmissionRequest
+}
+
+func (r *admissionTypeChecker) Reset() chan struct{} {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.upCh = make(chan struct{})
+	r.upOnce = sync.Once{}
+	r.requests = []*admissionv1.AdmissionRequest{}
+	return r.upCh
+}
+
+func (r *admissionTypeChecker) TypeCheck(req *admissionv1.AdmissionRequest, version string) *admissionv1.AdmissionResponse {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.requests = append(r.requests, req)
+	raw := req.Object.Raw
+	var into runtime.Object
+	if _, gvk, err := deserializerGVKTest.Decode(raw, nil, into); err != nil {
+		if gvk.Version != version {
+			return &admissionv1.AdmissionResponse{
+				UID:     req.UID,
+				Allowed: false,
+			}
+		}
+	}
+
+	return &admissionv1.AdmissionResponse{
+		UID:     req.UID,
+		Allowed: true,
+	}
+}
+
+func (r *admissionTypeChecker) MarkerReceived() {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.upOnce.Do(func() {
+		close(r.upCh)
+	})
+}
+
+func newAdmissionTypeCheckerHandler(recorder *admissionTypeChecker) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		data, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), 400)
+		}
+		review := admissionv1.AdmissionReview{}
+		if err := json.Unmarshal(data, &review); err != nil {
+			http.Error(w, err.Error(), 400)
+		}
+
+		switch r.URL.Path {
+		case "/marker":
+			recorder.MarkerReceived()
+			return
+		case "/v1":
+			review.Response = recorder.TypeCheck(review.Request, "v1")
+		case "/v2":
+			review.Response = recorder.TypeCheck(review.Request, "v2")
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		if err := json.NewEncoder(w).Encode(review); err != nil {
+			http.Error(w, err.Error(), 400)
+			return
+		}
+
+	})
+}
+
+// Test_MutatingWebhookConvertsGVKWithMatchPolicyEquivalent tests if a equivalent resource is properly converted between mutating webhooks
+func Test_MutatingWebhookConvertsGVKWithMatchPolicyEquivalent(t *testing.T) {
```

Comment:
I don't think conditions are active in this test, need this:

```
defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AdmissionWebhookMatchConditions, true)()
```

once you enable conditions, I suspect the test would fail until you make the condition changes noted below to check the actual apiVersion field

Reply:
ahh yeah you're right

Reply:
done